wdl-doc/src/callable.rs

```diff
@@ -0,0 +1,292 @@
+    fn required_inputs(&self) -> impl Iterator<Item = &Parameter> {
+        self.inputs().iter().filter(|param| {
+            param
+                .required()
+                .expect("inputs should return Some(required)")
+        })
+    }
+
+    /// Get the sorted set of unique `group` values of the inputs.
+    fn input_groups(&self) -> BTreeSet<String> {
+        self.inputs()
+            .iter()
+            .filter_map(|param| param.group().as_ref().map(|s| s.to_string()))
+            .collect()
+    }
+
+    /// Get the inputs of the callable that are part of `group`.
+    fn inputs_in_group<'a>(&'a self, group: &'a str) -> impl Iterator<Item = &'a Parameter> {
+        self.inputs().iter().filter(move |param| {
+            if let Some(param_group) = param.group() {
+                if param_group == group {
+                    return true;
+                }
+            }
+            false
+        })
+    }
+
+    /// Get the inputs of the callable that are neither required nor part of a
+    /// group.
+    fn other_inputs(&self) -> impl Iterator<Item = &Parameter> {
+        self.inputs().iter().filter(|param| {
+            !param
+                .required()
+                .expect("inputs should return Some(required)")
+                && param.group().is_none()
+        })
+    }
+
+    /// Render the required inputs of the callable.
+    fn render_required_inputs(&self) -> Markup {
+        let mut iter = self.required_inputs().peekable();
+        if iter.peek().is_some() {
+            return html! {
+                h3 { "Required Inputs" }
+                table class="border" {
+                    thead class="border" { tr {
+                        th { "Name" }
+                        th { "Type" }
+                        th { "Description" }
+                        th { "Additional Meta" }
+                    }}
+                    tbody class="border" {
+                        @for param in iter {
+                            (param.render())
+                        }
+                    }
+                }
+            };
+        };
+        html! {}
+    }
+
+    /// Render the common inputs of the callable.
+    fn render_common_inputs(&self) -> Markup {
+        let mut iter = self.inputs_in_group("Common").peekable();
+        if iter.peek().is_some() {
+            return html! {
+                h3 { "Common" }
+                table class="border" {
+                    thead class="border" { tr {
+                        th { "Name" }
+                        th { "Type" }
+                        th { "Default" }
+                        th { "Description" }
+                        th { "Additional Meta" }
+                    }}
+                    tbody class="border" {
+                        @for param in iter {
+                            (param.render())
+                        }
+                    }
+                }
+            };
+        };
+        html! {}
+    }
```

Comment:
Do we need to treat `Common` separately? Or can we apply a custom sort to `input_groups`?

Reply:
That should be possible. Another good thought.
I think if we're going down a custom sort order for `input_groupts`, `Common` should come first, and `Resources` should come last? Everything else would just be alphabetical between those two?